src/wattsi.pas

```diff
@@ -158,6 +165,90 @@    TCrossReferences = record
    CurrentSectionNumber: array[THeadingRank] of Cardinal;
    StringStore: TStringStore;
    Errors: Cardinal;
+   XrefsByDFNAnchor: TXrefsByDFNAnchor;
+
+   function NewXrefsByDFNAnchor(SectionName: UTF8String; Anchor: UTF8String): TXrefAnchorsBySectionName;
+   var
+      Anchors: TStringList;
+      XrefAnchorsBySectionName: TXrefAnchorsBySectionName;
+   begin
+      Anchors := TStringList.Create;
+      Anchors.Add(Anchor);
+      XrefAnchorsBySectionName := TXrefAnchorsBySectionName.Create;
+      XrefAnchorsBySectionName.Add(SectionName, Anchors);
+      Result := XrefAnchorsBySectionName;
+   end;
+
+   function NewXrefAnchorsBySectionName(SectionName: UTF8String; Anchor: UTF8String): TXrefAnchorsBySectionName;
+   var
+      Anchors: TStringList;
+      XrefAnchorsBySectionName: TXrefAnchorsBySectionName;
+   begin
+      Anchors := TStringList.Create;
+      Anchors.Add(Anchor);
+      XrefAnchorsBySectionName := TXrefAnchorsBySectionName.Create;
+      XrefAnchorsBySectionName.Add(SectionName, Anchors);
+      Result := XrefAnchorsBySectionName;
+   end;
+
+   procedure XrefsToJSON(XrefsByDFNAnchor: TXrefsByDFNAnchor);
```

Comment:
> FWIW it looks like there might be a way to use a package for this: http://wiki.freepascal.org/fcl-json. However I'm not sure it's worth the effort, especially if adding a dependency is nontrivial.

Yeah I already looked into using that but along with requiring multiple additional depedencies, it also requires restructuring the code in a way that creates objects in a way the library requires/expects. It is not capable of working with arbitrary objects that use normal Pascal sets/lists/maps. So I definitely think it’s not worth exploring further.

And given that we know the exact structure of the object we need to produce JSON from, it seems like the lightweight ad-hoc serializer in this PR is sufficient for the need at hand.